common/src/main/java/com/linecorp/centraldogma/internal/api/v1/CreateProjectRequest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.internal.api.v1;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+
+/**
+ * A request to create a new project.
+ */
+public class CreateProjectRequest {
+
+    private final String name;
+    private final List<String> owners;
+    private final List<String> members;
+
+    @JsonCreator
+    public CreateProjectRequest(@JsonProperty("name") String name,
+                                @JsonProperty("owners") @Nullable List<String> owners,
+                                @JsonProperty("members") @Nullable List<String> members) {
+        checkArgument(!isNullOrEmpty(name), "Project name should not be null or empty.");
```

Comment:
How about adding `Util.validateProjectName()` and `validateRepositoryName()`? You can find the rule at `DirectoryBasedStorageManager.CHILD_NAME`.